src/python/grpcio_tests/tests_aio/unit/server_interceptor_test.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2020 The gRPC Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+import unittest
+from typing import Callable
+
+import grpc
+
+from grpc.experimental import aio
+
+from tests_aio.unit._test_server import start_test_server
+from tests_aio.unit._test_base import AioTestBase
+from src.proto.grpc.testing import messages_pb2
+
+
+class _LoggingInterceptor(aio.ServerInterceptor):
+
+    def __init__(self, tag, record):
+        self.tag = tag
+        self.record = record
+
+    async def intercept_service(self, continuation, handler_call_details):
+        self.record.append(self.tag + ':intercept_service')
+        return await continuation(handler_call_details)
+
+
+class _GenericInterceptor(aio.ServerInterceptor):
+
+    def __init__(self, fn):
+        self._fn = fn
+
+    async def intercept_service(self, continuation, handler_call_details):
+        return await self._fn(continuation, handler_call_details)
+
+
+def _filter_server_interceptor(condition: Callable,
```

Comment:
I missed this one in my last round of review. By adding type annotation, I mean what is the argument it accepts and returns. You can do this like:

```python
def atoi(a: str) -> int:
    return int(a)

atoi: Callable[[str], int]
```

Also, if possible, please annotate all other arguments in this file, not only this function.

Reply:
OK